src/js/smooch.jsx

```diff
@@ -196,6 +195,40 @@ export class Smooch {
         });
     }
 
+    monitorUrlChanges() {
+        const onhashchange = window.onhashchange;
+        const pushState = window.history.pushState;
+        const replaceState = window.history.replaceState;
+
+        window.onhashchange = function() {
```

Comment:
> Can you do it for pushState? 

Actually you can listen to the popstate event on `window` (https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_popstate_event) instead of listening to replaceState and pushState

then you would window.addEventListener for both hashchange and popstate.